File: UI/ScriptBeeUI/src/app/types/project.ts
export interface Project {
  id: string;
  name: string;
  creationDate: string;
}

export interface ProjectListResponse {
  projects: Project[];
}

export interface CreateProjectRequest {
  id: string;
  name: string;
}

export interface CreateProjectResponse {
  id: string;
  name: string;
  creationDate: string;
}

File: ScriptBeeWebApp/src/Common/Model/Context/Project.cs
using DxWorks.ScriptBee.Plugin.Api.Model;

namespace ScriptBee.Domain.Model.Context;

public class Project : IProject
{
    public IContext Context { get; set; } = new Context();
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public DateTimeOffset CreationDate { get; set; }
}
